packages/fast-components-react-msft/src/action-toggle/action-toggle.spec.tsx

```diff
@@ -0,0 +1,291 @@
+        actionToggle__selected: "action-toggle-selected",
+        actionToggle__hasGlyphAndContent: "action-toggle-hasGlyphAndContent",
+    };
+    const href: string = "#";
+
+    test("should have a displayName that matches the component name", () => {
+        expect((MSFTActionToggle as any).name).toBe(MSFTActionToggle.displayName);
+    });
+
+    test("should not throw if managedClasses are not provided", () => {
+        expect(() => {
+            shallow(
+                <MSFTActionToggle selectedLabel="selected" unselectedLabel="unselected" />
+            );
+        }).not.toThrow();
+    });
+
+    test("should create unselectedglyph element passed in", () => {
+        const props: ActionToggleHandledProps = {
+            selectedLabel: "selected",
+            unselectedLabel: "unselected",
+            selectedGlyph: (className?: string): React.ReactNode => {
+                return <div>X</div>;
+            },
+            unselectedGlyph: (className?: string): React.ReactNode => {
+                return <div>Y</div>;
+            },
+        };
+
+        const rendered: any = mount(<ActionToggle {...props} />);
+
+        expect(rendered.contains(<div>Y</div>)).toEqual(true);
+    });
+
+    test("should create selectedglyph element passed in", () => {
+        const props: ActionToggleHandledProps = {
+            selectedLabel: "selected",
+            unselectedLabel: "unselected",
+            selectedGlyph: (className?: string): React.ReactNode => {
+                return <div>X</div>;
+            },
+            unselectedGlyph: (className?: string): React.ReactNode => {
+                return <div>Y</div>;
+            },
+            selected: true,
+        };
+
+        const rendered: any = mount(<ActionToggle {...props} />);
+
+        expect(rendered.contains(<div>X</div>)).toEqual(true);
+    });
+
+    test("should implement unhandledProps", () => {
+        const handledProps: ActionToggleProps = {
+            managedClasses,
+            href,
+            selectedLabel: "selected",
+            unselectedLabel: "unselected",
+        };
+
+        const unhandledProps: ActionToggleUnhandledProps = {
+            "aria-label": "label",
+        };
+
+        const props: ActionToggleProps = { ...handledProps, ...unhandledProps };
+
+        const rendered: any = mount(<ActionToggle {...props} />);
+
+        expect(rendered.first().prop("aria-label")).toEqual("label");
+    });
+
+    test("should apply a 'disabled' html class when prop disabled is passed", () => {
+        const props: ActionToggleHandledProps = {
+            selectedLabel: "selected",
+            unselectedLabel: "unselected",
+            disabled: true,
+        };
+
+        const rendered: any = mount(<ActionToggle {...props} />);
+
+        expect(rendered.find("button").prop("className")).toContain(
+            "actionToggle__disabled"
+        );
+    });
+
+    test("should apply a custom class-name", () => {
```

Comment:
nit: prefer "should add a custom class name if a class name has been provided"

Reply:
+1 also class name is two words

Reply:
Not seeing resolution or reasoning for not resolving

Reply:
Updated with changes.